content/docs/guides/error.md

```diff
@@ -18,6 +18,37 @@ If an error occurs, gRPC returns one of its error status codes instead, with an
 optional string error message that provides further details about what happened.
 Error information is available to gRPC clients in all supported languages.
 
+### Advanced error modeling
+
+The error model described above is the official gRPC error model,
+is supported by all gRPC client/server libraries, and is independent of
+the gRPC data format (whether protocol buffers or something else). You
+may have noticed that it's quite limited and doesn't include the
+ability to communicate error details.
+
+If you're using protocol buffers as your data format, however, many of
+the [gRPC language libraries](https://grpc.io/docs/reference/) now
+support the richer error model developed and used by Google as
+described
+[here](https://cloud.google.com/apis/design/errors#error_model). *TODO:
+this [SO post](https://stackoverflow.com/a/48750825) states that
+"Support is coming to each language to handle the type". What can be
+said here about that commitment and status/timeline?*
+
+Even if you're not using protocol buffers, you can similarly use gRPC
+response metadata to extend the standard error response with error
+details. However, you'd likely need to find or develop library support
+for accessing this data in order to make practical use of it in your
+APIs.
+
+One thing to be aware of if you adopt such an extended error model,
+however, is that existing proxies, loggers, and other standard HTTP
+request processors don't have visibility into the error details and
+thus wouldn't be able to leverage them for monitoring or other
+purposes.
+
+*TODO What other caveats, if any, should be mentioned here?*
```

Comment:
Big one:  Additional error detail in the trailers interferes with head-of-line blocking, and will cause large cache misses in the HPACK table.  Also, larger error bodies sometimes run into protocol limits (like max headers size), effectively losing the original error.  We have seen teams putting 8K of errors in each response.

Reply:
@carl-mastrangelo Added this, thanks.